content/en/blog/_posts/2024-05-13-gateway-api-v1.1/index.md

```diff
@@ -0,0 +1,256 @@
+have been run and provide reproduction steps.
+
+
+### New additions to Experimental channel
+
+#### [Gateway Client Certificate Verification](https://gateway-api.sigs.k8s.io/geps/gep-91/)
+
+Gateways can now configure client cert verification for each Gateway Listener by
+introducing a new `frontendValidation` field within `tls`. This field
+supports configuring a list of CA Certificates that can be used as a trust
+anchor to validate the certificates presented by the client.
+
+The following example shows how the CACertificate stored in
+the `foo-example-com-ca-cert` ConfigMap can be used to validate the certificates
+presented by clients connecting to the `foo-https` Gateway Listener.
+
+```
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: client-validation-basic
+spec:
+  gatewayClassName: acme-lb
+  listeners:
+    name: foo-https
+    protocol: HTTPS
+    port: 443
+    hostname: foo.example.com
+  tls:
+    certificateRefs:
+      kind: Secret
+      group: ""
+      name: foo-example-com-cert
+    frontendValidation:
+      caCertificateRefs:
+        kind: ConfigMap
+        group: ""
+        name: foo-example-com-ca-cert
+```
+
+#### [Session Persistence and BackendLBPolicy](https://gateway-api.sigs.k8s.io/geps/gep-1619/)
+
+[Session Persistence](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1.SessionPersistence) is being introduced to Gateway API via a new policy
+([BackendLBPolicy](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1alpha2.BackendLBPolicy))
+for Service-level configuration and as fields within HTTPRoute
+and GRPCRoute for route-level configuration. The BackendLBPolicy and route-level
+APIs provide the same session persistence configuration, including session
+timeouts, session name, session type, and cookie lifetime type.
+
+Below is an example configuration of `BackendLBPolicy` that enables cookie-based
+session persistence for the `foo` service. It sets the session name to
+`foo-session`, defines absolute and idle timeouts, and configures the cookie to
+be a session cookie:
+
+```
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: BackendLBPolicy
+metadata:
+  name: lb-policy
+  namespace: foo-ns
+spec:
+  targetRefs:
+  - group: core
+    kind: service
+    name: foo
+  sessionPersistence:
+    sessionName: foo-session
+    absoluteTimeout: 1h
+    idleTimeout: 30m
+    type: Cookie
+    cookieConfig:
+      lifetimeType: Session
+```
+
+### Everything else
+
+#### [TLS Terminology Clarifications](https://gateway-api.sigs.k8s.io/geps/gep-2907/)
+
+As part of a broader goal of making our TLS terminology more consistent
+throughout the API, we've introduced some breaking changes to BackendTLSPolicy.
+This has resulted in a new API version (v1alpha3) and will require any existing
+users of this policy to uninstall the v1alpha2 version before installing this
```

Comment:
Potentially:
```suggestion
users of this policy to **uninstall** the v1alpha2 version, before you install this
```

I'm inferring people also need to delete their BackendTLSPolicies before they delete the CRD; have I got that right? If so, let's be more explicit about that.

Reply:
> I'm inferring people also need to delete their BackendTLSPolicies before they delete the CRD; have I got that right? 

No need, deleting the CRD will also delete any CRs based on it.

Reply:
What I'm indirectly wondering: could there be a finalizer (baked in, not something customizer) that prevents or delays removal of a particular BackendTLSPolicy? If so, I would call that out here.

Reply:
So here's the (more) complete story. (Note that in the below, I'm using "CRD" to mean literally the CustomResourceDefinition resource that tells the API server how to handle custom resources, and "CR" to refer to instances of the custom resources thus defined. We often use those interchangeably, but it matters here.)

- If you just straight up delete the BackendTLSPolicy CRD, the API server will instantly and silently remove any BackendTLSPolicy CRs from the cluster, with no confirmation needed and no quarter given.
- If, on the other hand, your cluster has a `v1alpha2` BackendTLSPolicy CRD in place, and you attempt to apply a BackendTLSPolicy that no longer serves `v1alpha2` but instead serves `v1alpha3`, there is an interlock in the API server that will not accept the new CRD until you have explicitly told the API server that no `v1alpha2` CRs are in use. (You can lie to it, of course, and tell it that you're OK deleting `v1alpha2` when it's really not, but in that case on your own head be it.)

We haven't included the above in the docs because:

- the upshot is "installing `v1alpha3` requires you to take explicit action to tell the system that it's OK", which we _do_ include in the docs;
- _which_ explicit action you take depends on how you want to approach things; and
- we had previously thought that this was pretty well covered by the existing CRD documentation.

So, question back to you: do you think we're wrong there and we need to go into it more? or possibly write something else that goes into more detail, to which we then could link?

Reply:
I don't think the proposed text is wrong, but I did want to check that you think this particular detail looks right.

Some controllers set finalizers that make the story more complicated (imagine if a BackendTLSPolicy could have a finalizer, and this might indirectly block removal of the CRD). It doesn't sound like that's the case here.

Reply:
🤦‍♂️ Ugh, sorry for misunderstanding and explaining what you already knew, then. Correct – finalizers shouldn't be in play for BackendTLSPolicy.

Reply:
(no worries!)

Reply:
At the moment there are only one or two implementations of the experimental BackendTLSPolicy and so the meaning of "users" in this context arguably refers to implementations.  I will change this to:

> This has resulted in a new API version (v1alpha3) and will require any existing
> implementations of this policy to properly handle the version upgrade, e.g.
> by backing up data and uninstalling the v1alpha2 version before installing this
> newer version.